src/vision/status_quo/alan_creates_a_hanging_alarm.md

```diff
@@ -0,0 +1,129 @@
+# 😱 Status quo stories: Alan Creates a Hanging Alarm
+
+## 🚧 Warning: Draft status 🚧
+
+This is a draft "status quo" story submitted as part of the brainstorming period. It is derived from real-life experiences of actual Rust users and is meant to reflect some of the challenges that Async Rust programmers face today.
+
+If you would like to expand on this story, or adjust the answers to the FAQ, feel free to open a PR making edits (but keep in mind that, as they reflect peoples' experiences, status quo stories [cannot be wrong], only inaccurate). Alternatively, you may wish to [add your own status quo story][htvsq]!
+
+## The story
+
+[Alan] is a developer on the Bottlerocket project.
+[Bottlerocket] is a Linux-based open-source operating system that is purpose-built by Amazon Web Services for running containers.
+We use our own program [pubsys] to ensure that Bottlerocket update repositories are healthy.
+
+### The TL;DR - What makes Alan Sad
+
+Although the first two of these points are about tokio, they are really about Rust async since tokio serves as the de-facto `std::runtime` for Rust.
+
+- It is confusing and dangerous that multiple tokio runtimes can panic or hang at program runtime.
+- It is challenging that using multiple major versions of tokio (which is allowed by cargo) can fail at runtime.
+- It is unfortunate that we need a 3rd party runtime in order to `block_on` a future, even if we are not trying to write async code.
+
+### Multiple Tokio Runtimes
+
+Bottlerocket uses its own [tough] library to read and write TUF repositories.
+This library was created before async became widespread and [reqwest] changed its main interface to `async`.
+When reqwest switched to async, Alan used the `reqwest::blocking` feature instead of re-writing tough to be an async interface.
+(Maybe Alan [should](https://github.com/awslabs/tough/issues/213) make tough an async interface, but he hasn't yet.)
+In order to provide a non-async interface, `reqwest::blocking` creates a tokio runtime so that it can await futures.
+
+In pubsys Alan created some parallel downloading logic while using the above libraries.
```

Comment:
What is pubsys?

Reply:
Does this mean that Alan created parallel threads that use `reqwest`?

Reply:
I think I would say something like:

In pubsys, Alan spawned multiple tokio tasks, each of which performed downloads using reqwest:

```rust
tokio::spawn(async move {
    // do the reqwest logic
});
```

Reply:
Your example is exactly correct. I added the code example slightly closer to actual. If it is helpful at all, I looked up all the exact examples

Exact code references:
 - [pubsys](https://github.com/bottlerocket-os/bottlerocket/blob/fee7e7520254f5df5958e361245efcbee05b0b18/tools/pubsys/src/repo/validate_repo/mod.rs#L57)
 - [tough using reqwest::blocking](https://github.com/awslabs/tough/blob/f9f5549accfc672d0099f4a125b11d56c73f1a1c/tough/src/http.rs#L5)
 - [reqwest::blocking creating a runtime](https://github.com/seanmonstar/reqwest/blob/bd9ff9f371b756decf26fbbde1687433b0f63774/src/blocking/client.rs#L743) (though I'm not sure this line is the exact culprit)

However, I think the snippet encapsulates what happened well enough without digging into the actual code.